include/zephyr/drivers/led_strip.h

```diff
@@ -108,20 +108,19 @@ static inline int led_strip_update_rgb(const struct device *dev,
 				       struct led_rgb *pixels,
 				       size_t num_pixels)
 {
-	const struct led_strip_driver_api *api =
-		(const struct led_strip_driver_api *)dev->api;
+	__ASSERT_NO_MSG(DEVICE_API_IS(led_strip, dev));
 
 	/* Allow for out-of-tree drivers that do not have this function for 2 Zephyr releases
 	 * until making it mandatory, function added after Zephyr 3.6
 	 */
-	if (api->length != NULL) {
+	if (DEVICE_API_GET(led_strip, dev)->length != NULL) {
 		/* Ensure supplied pixel size is valid for this device */
-		if (api->length(dev) < num_pixels) {
+		if (DEVICE_API_GET(led_strip, dev)->length(dev) < num_pixels) {
 			return -ERANGE;
 		}
 	}
 
-	return api->update_rgb(dev, pixels, num_pixels);
+	return DEVICE_API_GET(led_strip, dev)->update_rgb(dev, pixels, num_pixels);
```

Comment:
This notation is quite heavy. Since we are using it several times, couldn't we just get the api pointer ?